File: datasets/jl-keep-code_davinci_001_temp_0.2/HumanEval_14_all_prefixes.jl
""" Return list of all prefixes from shortest to longest of the input string
    >>> all_prefixes('abc')
    ['a', 'ab', 'abc']
    """
function all_prefixes(string::String)::Vector{String} 
    prefixes = []
    for i in 1:length(string)
        push!(prefixes, string[1:i])
    end
    return prefixes
end

using Test

@testset begin

candidate = all_prefixes;
	@test(candidate("") == Vector{String}([]))
	@test(candidate("asdfgh") == ["a", "as", "asd", "asdf", "asdfg", "asdfgh"])
	@test(candidate("WWW") == ["W", "WW", "WWW"])
end

File: datasets/pl-keep-code_davinci_001_temp_0.2/HumanEval_14_all_prefixes.pl
# Return list of all prefixes from shortest to longest of the input string
# >>> all_prefixes('abc')
# ['a', 'ab', 'abc']
sub all_prefixes {
    my($string) = @_;
    my @prefixes = ();
    my $prefix = '';
    foreach my $char (split //, $string) {
        $prefix .= $char;
        push @prefixes, $prefix;
    }
    return @prefixes;
}

use Test::Deep;


sub testhumaneval {
    my $candidate = \&all_prefixes;
        if(eq_deeply($candidate->(""),[])) {
        print "ok!" }else{
        exit 1;
        }
        if(eq_deeply($candidate->("asdfgh"),["a", "as", "asd", "asdf", "asdfg", "asdfgh"])) {
        print "ok!" }else{
        exit 1;
        }
        if(eq_deeply($candidate->("WWW"),["W", "WW", "WWW"])) {
        print "ok!" }else{
        exit 1;
        }
}

testhumaneval();